compose/service.py

```diff
@@ -1074,7 +1076,12 @@ def build(self, no_cache=False, pull=False, force_rm=False, memory=None, build_a
         )
 
         try:
-            all_events = stream_output(build_output, sys.stdout)
+            if not quiet:
+                all_events = stream_output(build_output, sys.stdout)
+            else:
+                all_events = []
```

Comment:
You can simplify this by using the same trick as in `service.pull`: https://github.com/docker/compose/blob/f207d94b3c1be9bf83bf5421d1c88043eeedc650/compose/service.py#L1190-L1196